Add section to integration document for additional `makePaymentRequest` parameters. (#631)
commercetools/commercetools-adyen-integration: extension/docs/WebComponentsIntegrationGuide.md

- `makePaymentRequest` must contain a unique payment `reference` value. The reference value cannot be duplicated in any commercetools payment and it's a required field by Adyen. The extension module uses the `reference` value to set the payment key, later it acts as a unique link between commercetools payment and Adyen payment(`merchantReference`). `Reference` may only contain alphanumeric characters, underscores and hyphens and must have a minimum length of 2 characters and a maximum length of 80 characters.
- `payment.amountPlanned` can not be changed if there is a `makePayment` interface interaction present in the commercetools payment object. The `amount` value in `makePaymentRequest` custom field must have the same value as `payment.amountPlanned`. This ensures eventual payment amount manipulations (i.e.: when [my-payments](https://docs.commercetools.com/http-api-projects-me-payments#my-payments) are used) for already initiated payment.

**Important**

> In this integration document our Adyen payment request examples are trimmed to minimum, for instance, a `makePaymentRequest` for 3D Secure 2 payment type, should have `additionalData.allow3DS2` set to `true`. So in your integration, you might need to include additional parameters in your payment request to:

Reviewer: ```suggestion
In this integration document our Adyen payment request examples are trimmed to minimum. Depending on your requirements you might want to include other Adyen parameters like (add risk management fields)[https://docs.adyen.com/risk-management/configure-standard-risk-rules/required-risk-field-reference], (activate 3D Secure 2)[https://docs.adyen.com/online-payments/3d-secure/native-3ds2/web-component#make-a-payment] or (allow recurring payments)[https://docs.adyen.com/payment-methods/cards/web-component#create-a-token]. Please find all the possible parameters in the `Web Components` section of the desired payment method listed in the navigation (here)[https://docs.adyen.com/payment-methods]
```

Author: https://github.com/commercetools/commercetools-adyen-integration/pull/631/commits/e6ace00d641f371ca636737899b2f0da5ec22fb8

> In this integration document our Adyen payment request examples are trimmed to minimum, for instance, a `makePaymentRequest` for 3D Secure 2 payment type, should have `additionalData.allow3DS2` set to `true`. So in your integration, you might need to include additional parameters in your payment request to:

Author: https://github.com/commercetools/commercetools-adyen-integration/pull/631/commits/e6ace00d641f371ca636737899b2f0da5ec22fb8

- `payment.amountPlanned` can not be changed if there is a `makePayment` interface interaction present in the commercetools payment object. The `amount` value in `makePaymentRequest` custom field must have the same value as `payment.amountPlanned`. This ensures eventual payment amount manipulations (i.e.: when [my-payments](https://docs.commercetools.com/http-api-projects-me-payments#my-payments) are used) for already initiated payment.

**Important**

In this integration document our Adyen payment request examples are trimmed to minimum. Depending on your requirements you might want to include other Adyen parameters such as [add risk management fields](https://docs.adyen.com/risk-management/configure-standard-risk-rules/required-risk-field-reference), [activate 3D Secure 2](https://docs.adyen.com/online-payments/3d-secure/native-3ds2/web-component#make-a-payment) or [allow recurring payments](https://docs.adyen.com/payment-methods/cards/web-component#create-a-token).
Please find all the possible parameters in the `Web Components` section of the desired payment method listed in the navigation [here](https://docs.adyen.com/payment-methods).

> Here's an example of the value of the `makePaymentRequest` custom field for 3D Secure credit cart payment , with generated component data from Adyen Web Components for a credit card payment:

Reviewer 2: I think we need `3D Secure 2` - 2 is important, as we are telling to the Adyen - our app is ready for 3ds2 handling. I am pretty sure, if it is not provided - it would force 3ds1 (redirect) if card's issuer requires it.

Also we could get rid of this space before `, with`

> Here's an example of the value of the `makePaymentRequest` custom field for 3D Secure 2 credit cart payment, with generated component data from Adyen Web Components for a credit card payment:

Reviewer: ```suggestion
Here's an example of the `makePaymentRequest` custom field value for 3D Secure 2 credit card payment along with the generated Adyen Web Components data:
```

    "encryptedExpiryYear": "test_2030",
    "encryptedSecurityCode": "test_737"
  },
  "additionalData": {
    "allow3DS2": true
  },

>   "channel": "Web",

Reviewer 2: I am not sure if it is case sensitive or not in Adyen. However, I've used lowercase - `web` and I can confirm this case works fine - not sure about `Web`

Reviewer 2: If you decide to change it, please remember there are 3 places below to update too

Author: It's "Web" in the Adyen docs, probably ok to keep as is.
